### PR TITLE
feat: Add OpenAI support

### PR DESCRIPTION
**Pull request by kun432**
This PR adds support for OpenAI.

SS_ 2024-09-05 8 34 41

SS_ 2024-09-05 8 25 33

SS_ 2024-09-05 9 01 13

### CODE REVIEW

**Comment by kun432** [comment]
one thing I noticed.

The models API is being requested every 5 seconds. Isn't that a bit too frequent?

https://github.com/gyopak/sidellama/blob/master/src/sidePanel/hooks/useUpdateModels.ts#L65

**Comment by gyopak** [comment]
hey @kun432 

This is amazing, thanks for the contribution! afaik in general providers don't bill after /models API calls, but yeah, 5sec is too frequent. Actually it is causing a rendering bug, the model selector rerenders every 5th secs. I'm working on a minor release with bug fixes, I'll make sure to include an adjustment for the model gathering logic.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sidellama",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (gyopak) [inline package.json]: Please remove the bump. `0.0.2` will be released soon including this feature and various bug fixes.

Comment (kun432) [reply package.json]: done

**Comment by kun432** [comment]
@gyopak Thank you!

**Pull request merged**